### PR TITLE
Add cloudbuild to copy data from repo to GCS for custom DC

### PR DESCRIPTION
**Pull request by shifucun**

### CODE REVIEW

File: copydata.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+declare -A foldermap
+foldermap["scripts/stanford"]="gs://datcom-stanford-resources/imports"
+
+
+changes() {
+  git diff --name-only --diff-filter=AMDR --cached HEAD^ $1
+}
+
+for folder in ${!foldermap[@]}; do
```
Comment (pradh) [inline copydata.sh:29]: Maps in bash are tricky to follow, maybe have some comments on what it is expected to do? :)

File: copydata.sh [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+# A list of folders that we watch for data update and copy to GCS in Custom DC
+# The layout under each folder should match the storage layout in
+# https://docs.datacommons.org/custom_dc/upload_data.html
+
+declare -A foldermap
+foldermap["scripts/stanford"]="gs://datcom-stanford-resources/imports"
+
+
+changes() {
+  git diff --name-only --diff-filter=AMDR --cached HEAD^ $1
```
Comment (pradh) [inline copydata.sh:26]: Add a comment about what this does?

File: copydata.sh [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+for folder in ${!foldermap[@]}; do
+  gcs_path=${foldermap[$folder]}
+  if [[ $(changes "$folder") == "" ]];  then
```
Comment (pradh) [inline copydata.sh:31]: hmm.. should this be != ""?

File: copydata.sh [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    echo "Copy data to $gcs_path"
+    for dir in $(find $folder -type d -maxdepth 1 -mindepth 1)
+    do
+      import_group=$(basename $dir)
+      echo $import_group
+      gsutil cp -r "$folder/$import_group/data/*" "$gcs_path/$import_group/data/"
```
Comment (pradh) [inline copydata.sh:37]: Add TODO to handle GCS references?

Comment (pradh) [inline copydata.sh:37]: nit: maybe have a set -x at the top so we can follow along and debug easily if needed?

**Review by pradh — COMMENTED** [review]
Very nice!

File: copydata.sh (hunk 2 above)
Comment (shifucun) [reply copydata.sh:26]: Done

File: copydata.sh (hunk 1 above)
Comment (shifucun) [reply copydata.sh:29]: Added a lot of comments.

File: copydata.sh (hunk 3 above)
Comment (shifucun) [reply copydata.sh:31]: Yes! This was used for testing.. reverted.

File: copydata.sh (hunk 4 above)
Comment (shifucun) [reply copydata.sh:37]: Done

Comment (shifucun) [reply copydata.sh:37]: Done

**Review by pradh — APPROVED** [review]
Thanks for the comments

File: copydata.sh [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+
+set -x
+
+# Define a map from folder path under this repo, to the GCS path to copy data to.
+declare -A foldermap
+foldermap["scripts/stanford"]="gs://datcom-stanford-resources/imports"
+
+
+# This function run "git diff" between latest and the previosu commit on the
```
Comment (pradh) [inline copydata.sh:29]: nit: previous

File: copydata.sh [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  gcs_path=${foldermap[$folder]}
+  # If the current commit touches files in the target folder, copy data.
+  if [[ $(changes "$folder") != "" ]];  then
+    echo "Copy data to $gcs_path"
+    for dir in $(find $folder -type d -maxdepth 1 -mindepth 1)
+    do
+      import_group=$(basename $dir)
+      echo $import_group
+      # Should only copy content under /data folder for each import group.
+      # In GCS, there are /internal folders that should not be overwrite.
```
Comment (pradh) [inline copydata.sh:47]: nit: overwritten

File: copydata.sh (hunk 5 above)
Comment (shifucun) [reply copydata.sh]: Done

File: copydata.sh (hunk 6 above)
Comment (shifucun) [reply copydata.sh]: Done

**Review by shifucun — COMMENTED** [review]
thanks!

**Pull request merged**